File: ios/LiquidGlassViewManager.m
/**
 * LiquidGlassViewManager.m — React Native View Manager Bridge
 *
 * Objective-C bridge for LiquidGlassViewManager Swift class.
 * Exports view properties to React Native.
 *
 * @see LiquidGlassViewManager.swift for Swift implementation
 */

#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(LiquidGlassViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(tintColorHex, NSString)
RCT_EXPORT_VIEW_PROPERTY(tintIntensity, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(glassStyle, NSString)
RCT_EXPORT_VIEW_PROPERTY(cornerRadius, CGFloat)
RCT_EXPORT_VIEW_PROPERTY(fallbackColorHex, NSString)

@end

File: ios/LiquidGlassViewManager.swift
/**
 * LiquidGlassViewManager — React Native View Manager for Liquid Glass
 *
 * Bridges LiquidGlassNativeView to React Native.
 *
 * @see LiquidGlassModule.swift for the actual view implementation
 * @see LiquidGlassViewManager.m for Objective-C bridge
 * @see .claude/plans/LIQUID_GLASS_IMPLEMENTATION.md
 */

import React

// ============================================================
// MARK: - LiquidGlassViewManager
// ============================================================

@objc(LiquidGlassViewManager)
class LiquidGlassViewManager: RCTViewManager {

    override func view() -> UIView! {
        return LiquidGlassNativeView()
    }

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
